lib/hdf5-db/jhdf5-db-shared.c

```diff
@@ -120,6 +120,15 @@ H5VL_julea_db_object_unref(JHDF5Object_t* object)
 		{
 			case J_HDF5_OBJECT_TYPE_FILE:
 				g_free(object->file.name);
+
+				// root group ressources can be directly freed here
+				if (object->file.root_group)
+				{
+					g_free(object->file.root_group->group.name);
+					g_free(object->file.root_group->backend_id);
+					g_free(object->file.root_group);
+				}
```

Comment:
This code is only reached when the file is about to be freed.
Calling object_unref for the root group here would attempt to double free the file.
The problem is related to the reference count mentioned above:
If the file holds a reference to the root group and the root group to the file they will never be freed by object_unref because both will never reach the switch statement.

Another solution would be to introduce a special case for this in the object_unref function.